src/cmd/linuxkit/azure.go

```diff
@@ -0,0 +1,372 @@
+	}
+
+	return &subnet
+}
+
+func createPublicIPAddress(resourceGroup resources.Group, ipName, location string) *network.PublicIPAddress {
+	fmt.Printf("\nCreating public IP Address in resource group %s, with name %s", *resourceGroup.Name, ipName)
+
+	ipParameters := network.PublicIPAddress{
+		Location: &location,
+		PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
+			DNSSettings: &network.PublicIPAddressDNSSettings{
+				DomainNameLabel: to.StringPtr(ipName),
+			},
+		},
+	}
+	ipAddressChannel, errorChannel := publicIPAddressesClient.CreateOrUpdate(*resourceGroup.Name, ipName, ipParameters, nil)
+	for {
+		select {
+		case _, ok := <-ipAddressChannel:
+			if !ok {
+				ipAddressChannel = nil
+			}
+		case _, ok := <-errorChannel:
+			if !ok {
+				errorChannel = nil
+			}
+		}
+		if ipAddressChannel == nil && errorChannel == nil {
+			break
+		}
+	}
+	time.Sleep(time.Second * 5)
+	publicIPAddress, err := publicIPAddressesClient.Get(*resourceGroup.Name, ipName, "")
+	if err != nil {
+		fmt.Println(err.Error())
+		log.Fatalf("Unable to retrieve public IP address")
+	}
+	return &publicIPAddress
+}
+
+func createNetworkInterface(resourceGroup resources.Group, networkInterfaceName string, publicIPAddress network.PublicIPAddress, subnet network.Subnet, location string) *network.Interface {
+
+	networkInterfaceParameters := network.Interface{
+		Location: &location,
+		InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
+			IPConfigurations: &[]network.InterfaceIPConfiguration{
+				{
+					Name: to.StringPtr(fmt.Sprintf("IPconfig-%s", networkInterfaceName)),
+					InterfaceIPConfigurationPropertiesFormat: &network.InterfaceIPConfigurationPropertiesFormat{
+						PublicIPAddress:           &publicIPAddress,
+						PrivateIPAllocationMethod: network.Dynamic,
+						Subnet: &subnet,
+					},
+				},
+			},
+		},
+	}
+	networkInterfaceChannel, errorChannel := interfacesClient.CreateOrUpdate(*resourceGroup.Name, networkInterfaceName, networkInterfaceParameters, nil)
+	for {
+		select {
+		case _, ok := <-networkInterfaceChannel:
+			if !ok {
+				networkInterfaceChannel = nil
+			}
+		case _, ok := <-errorChannel:
+			if !ok {
+				errorChannel = nil
+			}
+		}
+		if networkInterfaceChannel == nil && errorChannel == nil {
+			break
+		}
+	}
+
+	networkInterface, err := interfacesClient.Get(*resourceGroup.Name, networkInterfaceName, "")
+	if err != nil {
+		log.Fatalf("Unable to retrieve network interface")
+	}
+	return &networkInterface
+}
+
+func setVirtualMachineParameters(storageAccountName string, networkInterfaceID, location string) compute.VirtualMachine {
+	return compute.VirtualMachine{
+		Location: &location,
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			HardwareProfile: &compute.HardwareProfile{
+				VMSize: compute.StandardDS1,
+			},
+			OsProfile: &compute.OSProfile{
+				ComputerName:  to.StringPtr("linuxkit"),
+				AdminUsername: to.StringPtr("dummyusername"),
```

Comment:
Just to be clear: those parameters are only specified so that the deployment gets validated. Since there is no Azure Linux Agent, those values will **not** be enforced and users can only connect if they included the SSHD service.

What do you think should be the way to go here?